pkg/pillar/agentbase/agent.go

```diff
@@ -13,51 +15,125 @@ import (
 // Each agent is expected to have a Context that implements
 // this interface.
 type AgentBase interface {
+	// AgentBaseContext - returns a pointer to the agentbase context
 	AgentBaseContext() *Context
-	AddAgentSpecificCLIFlags()
-	ProcessAgentSpecificCLIFlags()
+}
+
+const (
+	// DefaultErrorTime - the default error time
+	DefaultErrorTime = 3 * time.Minute
+	// DefaultWarningTime - the default warning time
+	DefaultWarningTime = 40 * time.Second
+	defaultLogLevel    = log.InfoLevel
+)
+
+// DefaultContext - returns a context with default values
+func DefaultContext(agentName string) Context {
+	return Context{
+		AgentName:   agentName,
+		WarningTime: DefaultWarningTime,
+		ErrorTime:   DefaultErrorTime,
+		AgentOptions: AgentOptions{
+			NeedWatchdog:          true,
+			CheckAndCreatePidFile: true,
+			StartAgentLog:         true,
+			// Temporary default until all agents use this
+			SubscribeToGlobalConfig: false,
+			GlobalConfigSubscriptionOptions: pubsub.SubscriptionOptions{
+				AgentName:   "",
+				TopicImpl:   types.ConfigItemValueMap{},
+				Activate:    false,
+				WarningTime: DefaultWarningTime,
+				ErrorTime:   DefaultErrorTime,
+			},
+		},
+		GlobalConfig: types.DefaultConfigItemValueMap(),
+	}
 }
 
 // CliParams - stores all the common cli params
 type CliParams struct {
 	DebugOverride bool
+	Debug         bool
 }
 
+// AgentOptions - stores various options an agent may use to control different functions in Run()
+type AgentOptions struct {
+	NeedWatchdog                    bool
```

Comment:
I don't understand the relationship between NeedWatchdog and StillRunning; they have no characters in common. Yet the implementation seems to think they mean the same thing.
So perhaps this should be called NeedStillRunning